hack/kind-install-for-capd.sh

```diff
@@ -56,21 +60,40 @@ fi
 # https://github.com/kubernetes-sigs/kind/issues/2875
 # https://github.com/containerd/containerd/blob/main/docs/cri/config.md#registry-configuration
 # See: https://github.com/containerd/containerd/blob/main/docs/hosts.md
-cat <<EOF | kind create cluster --name="$KIND_CLUSTER_NAME"  --config=-
-kind: Cluster
-apiVersion: kind.x-k8s.io/v1alpha4
-networking:
-  ipFamily: ${KIND_NETWORK_IPFAMILY}
-nodes:
-- role: control-plane
-  extraMounts:
-    - hostPath: /var/run/docker.sock
-      containerPath: /var/run/docker.sock
-containerdConfigPatches:
-- |-
-  [plugins."io.containerd.grpc.v1.cri".registry]
-    config_path = "/etc/containerd/certs.d"
+
+if [[ "$KIND_INFRA_PROVIDER" == "kubevirt" ]]; then
+  cat <<EOF | kind create cluster --name="$KIND_CLUSTER_NAME"  --config=-
```

Comment:
Despite the name of this shell script I assume if infra provider is set to kubevirt this is not supposed to work with CAPD?
(docker mountpoint is not added)

Additionally KIND_NETWORK_IPFAMILY is not respected here

Reply:
>Despite the name of this shell script I assume if infra provider is set to kubevirt this is not supposed to work with CAPD?

Correct, I wanted to reuse the logic in this script with minimal changes. Do you think we should move the common logic to a new script which would be called by both `kind-install-for-capd.sh` and `kind-install-for-capk.sh`?

>Additionally KIND_NETWORK_IPFAMILY is not respected here

True. It wasn't clear to me why we might need IPv6 support and I thought IPv6 support might interfere with the logic we're adding for resolving the Docker IP prefix for the MetalLB config. I can add it if you think it's useful.

Reply:
Done.

Reply:
Thx